lib/data_stores/mysql_advanced_data_store.py

```diff
@@ -0,0 +1,636 @@
+#!/usr/bin/env python
+# -*- mode: python; encoding: utf-8 -*-
+
+"""An implementation of a data store based on mysql."""
+
+
+import Queue
+import threading
+import thread
+import time
+import MySQLdb
+import re
+
+from MySQLdb import cursors
+
+from grr.lib import aff4
+from grr.lib import config_lib
+from grr.lib import data_store
+from grr.lib import rdfvalue
+from grr.lib import utils
+from grr.lib.data_stores import common
+
+# pylint: disable=nonstandard-exception
+class Error(data_store.Error):
+  """Base class for all exceptions in this module."""
+# pylint: enable=nonstandard-exception
+
+
+class MySQLConnection(object):
```

Comment:
The connection and the pool should be reused between the two MySQL based data stores - the changes here are good, we should take advantage of them.